client/src/app/pages/waves/waves.tsx

```diff
@@ -0,0 +1,330 @@
+    filteredItems,
+    getSortValues
+  );
+
+  const { currentPageItems, setPageNumber, paginationProps } =
+    usePaginationState(sortedItems, 10);
+
+  const columnNames = {
+    select: "",
+    name: "Name",
+    startDate: "Start date",
+    endDate: "End date",
+    applications: "Applications",
+    stakeholders: "Stakeholders",
+    status: "Status",
+    actions: "",
+  };
+
+  //Compound expandable code
+  type ColumnKey = keyof typeof columnNames;
+
+  const [expandedCells, setExpandedCells] = React.useState<
+    Record<string, ColumnKey>
+  >({
+    wave1: "applications", // Default to the first cell of the first row being expanded
+  });
+  const setCellExpanded = (
+    application: Application,
+    columnKey: ColumnKey,
+    isExpanding = true
+  ) => {
+    const newExpandedCells = { ...expandedCells };
+    if (isExpanding) {
+      newExpandedCells[application.name] = columnKey;
+    } else {
+      delete newExpandedCells[application.name];
+    }
+    setExpandedCells(newExpandedCells);
+  };
+  const compoundExpandParams = (
+    application: Application,
+    columnKey: ColumnKey,
+    rowIndex: number,
+    columnIndex: number
+  ): TdProps["compoundExpand"] => ({
+    isExpanded: expandedCells[application.name] === columnKey,
+    onToggle: () =>
+      setCellExpanded(
+        application,
+        columnKey,
+        expandedCells[application.name] !== columnKey
+      ),
+    expandId: "composable-compound-expandable-example",
+    rowIndex,
+    columnIndex,
+  });
+  const handleIsRowSelected = (row: IComposableRow) => {
+    const matchingWave = waves.find((wave) => wave.name === row.name);
+    return matchingWave ? isRowSelected(matchingWave) : false;
+  };
+  const handleToggleRowSelected = (
+    row: IComposableRow,
+    isSelecting: boolean
+  ) => {
+    const matchingWave = waves.find((wave) => wave.name === row.name);
+    matchingWave && toggleRowSelected(matchingWave, isSelecting);
+  };
+  //RBAC
+  // xxxxWriteAccess = checkAccess(userScopes, waveWriteScopes);
+  const waveRowDropdownItems = true //TODO: Check RBAC access
+    ? [
+        <DropdownItem
+          key="bulk-export-to-issue-manager"
+          component="button"
+          // onClick={() => setExportIssueModalOpen(true)}
+        >
+          {t("actions.export")}
+        </DropdownItem>,
+        <DropdownItem
+          key="bulk-delete"
+          // onClick={() => {
+          // }}
+        >
+          {t("actions.delete")}
+        </DropdownItem>,
+      ]
+    : [];
+  //
+
+  const rows: IComposableRow[] = [];
+  currentPageItems?.forEach((item, index) => {
+    const expandedCellKey = expandedCells[item.name];
+    const isRowExpanded = !!expandedCellKey;
+    const isSelected = isRowSelected(item);
+    rows.push({
+      name: item.name,
+      isExpanded: isRowExpanded,
+      expandedCellKey: expandedCellKey,
+      isSelected: isSelected,
+      expandedContentMap: {
```

Comment:
@mturley This was my solution for supporting dynamic expandable content (nested tables inside the root table.

Reply:
I see! That makes sense.